packages/calypso-e2e/src/lib/components/site-select-component.ts

```diff
@@ -0,0 +1,67 @@
+import { Page } from 'playwright';
+import { getCalypsoURL } from '../../data-helper';
+
+const selectors = {
+	searchInput: '[aria-label="Search"]',
+	siteList: '.site-selector__sites',
+};
+
+/**
+ * Represents the Site Select component.
+ */
+export class SiteSelectComponent {
+	private page: Page;
+
+	/**
+	 * Creates an instance of the Site Select component.
+	 *
+	 * @param {Page} page Object representing the base page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Check if the Site Selector page is shown.
+	 *
+	 * @returns {Promise<boolean>} Whether the site selector is shown.
+	 */
+	async siteSelectorShown(): Promise< boolean > {
+		await this.page.waitForLoadState( 'load' );
+		return await this.page.isVisible( ':has-text("Select a site")' );
+	}
+
+	/**
+	 * Given a site URL, search for and select the matching site.
+	 *
+	 * This method will forcibly navigate the page to the /home endpoint with no site forming
+	 * part of the URL. The raw /home endpoint will present to the user a list of sites belonging
+	 * to the said user.
+	 *
+	 * This method will then enter the desired site URL to the input field, wait for the matching
+	 * result to load, then click on the entry.
+	 *
+	 * @param {string} url URL of the desired site. Must not include the protocol (https).
+	 * @returns {Promise<void>} No return value.
+	 */
+	async selectSite( url: string ): Promise< void > {
+		// Wait for any preceding page load to finish first.
+		await this.page.waitForLoadState( 'load' );
+		const homeURL = getCalypsoURL( 'home' );
+		await this.page.goto( homeURL );
+		// Navigate to the raw /home endpoint to trigger site selection.
+		await this.page.waitForLoadState( 'load' );
+
+		await this.page.fill( selectors.searchInput, url );
+		// For some accounts with many sites, the search process takes a looooong time.
+		await this.page.waitForSelector( '.is-loading', { state: 'hidden', timeout: 60 * 1000 } );
+
+		await Promise.all( [
+			this.page.waitForNavigation(),
+			this.page.click( `${ selectors.siteList } :text("${ url }")`, { timeout: 60 * 1000 } ),
+		] );
+
+		// Assert the resulting URL is in the form of <protocol><calypsoBaseURL>/home/<url>.
+		await this.page.waitForURL( `**/home/${ url }` );
+	}
```

Comment:
Optional question: do we want to simplify this method to just interacting with a site selector that is already loaded?

My guess is that folks will use this component when they know the site selector is on screen, and you provide a method to check that. So we probably don't need to do the initial home navigation.

If we want to provide a full method that forces site selection, then we could make a Flow that takes of doing the initial home navigation and then would call this simplified method?

Reply:
Good point, I thought about this while I was writing it and your argument makes sense.

I'll go ahead and split the existing `selectSite` to two methods:
- one that checks if the selector is shown
- one to perform the selection.

I think that would be better separation of concerns and would also trim down the size of the method a bit.